experimental/SetPartitions/src/Util.jl

```diff
@@ -0,0 +1,95 @@
+"""
+    helper_new_point_values_vector(p::Vector{Vector{Int}}, q::Vector{Vector{Int}})
+
+Return a semantically identical partition in form of an vector, 
+which has new number values.
+"""
+function helper_new_point_values_vector(p::Vector, q::Vector)
+
+    p_points = vcat(copy(p[1]), copy(p[2]))
+
+    if !(isempty(p_points))
+        new_id, index = findmax(p_points)
+        new_id += 1
+    else
+        new_id = 1 
+    end
+    
+    q_points = vcat(copy(q[1]), copy(q[2]))
+    new_ids = Dict()
+
+    for n in q_points
+        new_ids[n] = new_id
+        new_id += 1
+    end
+    upper = []
+    for (i, n) in enumerate(q[1])
+        push!(upper, get(new_ids, n, -1))
+    end
```

Comment:
You don't need `i`. And `upper` is a `Vector{Any}`. How about
```suggestion
    upper = [get(new_ids, n, -1) for n in q[1]]
```
or perhaps even this (I did not try to reason whether `new_ids[n]` will be defined in all cases):
```suggestion
    upper = [new_ids[n] for n in q[1]]
```